modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryAdapter.java

```diff
@@ -390,9 +390,10 @@ public int pageSize() {
 
     /** {@inheritDoc} */
     @Override public CacheQuery<T> timeout(long timeout) {
-        A.ensure(timeout >= 0, "timeout >= 0");
-
-        this.timeout = timeout;
+        if (timeout >= 0)
+            this.timeout = timeout;
+        else
+            this.timeout = cctx.gridConfig().getDefaultQueryTimeout();
```

Comment:
From the first glance changed behavior of this method looks counter intuitive: if a passed argument is lesser than 0 a default timeout from a configuration is used. For me a check employed before asserting that negative values are invalid.

And does this change really makes queries use a default timeout? I suppose a small test will answer this question.